api-catalog-services/src/main/java/org/zowe/apiml/apicatalog/staticapi/StaticAPIRefreshController.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+package org.zowe.apiml.apicatalog.staticapi;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/static-api")
+@RequiredArgsConstructor
+public class StaticAPIRefreshController {
+
+    private final StaticAPIService staticAPIService;
+
+    @PostMapping(value = "/refresh", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity refreshStaticApis() {
+        StaticAPIResponse staticAPIResponse = staticAPIService.refresh();
+        return ResponseEntity
```

Comment:
When static api parsing encounters errors, and the dc response is like so:
```json
{
  "errors": [
    {
      "messageTemplate": {
        "key": "org.zowe.apiml.discovery.errorParsingStaticDefinitionData",
        "number": "ZWEAD702",
        "type": "WARNING",
        "text": "Unable to process static API definition data: '%s' - '%s'",
        "reason": "A problem occurred while parsing a static API definition file.",
        "action": "Review the mentioned static API definition file for errors.\n Refer to the specific log message to see what is the exact cause of the problem:\n\n - ServiceId is not defined in the file '%s'. The instance will not be created. Make sure to specify the ServiceId.\n - The `instanceBaseUrls` parameter of %s is not defined. The instance will not be created. Make sure to specify the `InstanceBaseUrl` property.\n - The API Catalog UI tile ID %s is invalid. The service %s will not have an API Catalog UI tile. Specify the correct catalog title ID.\n - One of the instanceBaseUrl of %s is not defined. The instance will not be created. Make sure to specify the InstanceBaseUrl property.\n - The URL %s does not contain a hostname. The instance of %s will not be created. The specified URL is malformed. Make sure to specify valid URL.\n - The URL %s does not contain a port number. The instance of %s will not be created.\n - The specified URL is missing a port number. Make sure to specify a valid URL.\n - The URL %s is malformed. The instance of %s will not be created: The Specified URL is malformed. Make sure to specify a valid URL.\n - The hostname of URL %s is unknown. The instance of %s will not be created: The specified hostname of the URL is invalid. Make sure to specify valid hostname.\n - Invalid protocol. The specified protocol of the URL is invalid. Make sure to specify valid protocol.\n - Additional service metadata of %s in processing file %s could not be created: %s"
      },
      "convertedText": "Unable to process static API definition data: 'C:\\workspace\\prod\\api-layer\\..\\api-defs\\create-react-app - Copy.yml' - 'Cannot deserialize instance of `java.util.ArrayList` out of VALUE_STRING token\n at [Source: (StringReader); line: 4, column: 11] (through reference chain: org.zowe.apiml.discovery.staticdef.Definition[&quot;services&quot;])'"
    }
  ],
  "instances": [
    {
      "instanceId": "STATIC-localhost:staticclient:10012",
      "app": "STATICCLIENT",
      "appGroupName": null,
      "ipAddr": "127.0.0.1",
      "sid": "na",
      "homePageUrl": "https://localhost:10012/discoverableclient/",
      "statusPageUrl": "https://localhost:10012/discoverableclient/application/info",
```
the api catalog ui reports static api refresh success.

Do we intend to hide it from the user? I am admin, and i am making changes to static defs and using refresh button to reload them. I make mistakes, but ui tells me that refresh is ok, yet i don't see my new service and i don't know what's going on. This is what i imagine can happen.

Reply:
You're right, the thing is that I check for the error code which is comes from the backend and in this case it is 200. Should it maybe changed in backend side in case of `org.zowe.apiml.discovery.errorParsingStaticDefinitionData`?

Reply:
Or I can check whether there error has a body on frontend side. Just wondering which solution is better

Reply:
not sure there is great code to capture the state, as the request goes through, and it gets the resource. The operation relared information is in the request...

Reply:
Addressed in the last commit